### ISSUE TITLE
Imported heightmap has different shape from original object

### ISSUE THREAD

**Issue by Moonwards** [Zylann/godot_heightmap_plugin#90]
We thought this might be an issue with how we created the heightmap from the Blender file we want to copy, but after trying it three different ways and talking it over, I wonder if the add-on has interpreted the heightmap in some way that changes the shape produced. There are a bunch of differences between the mesh produced in Godot and the original object in Blender.

![unknown](https://user-images.githubusercontent.com/50932660/60620741-9fee4980-9da1-11e9-9b76-178af150e836.png)

This is the heightmap being used

![LalandeHeightMap2](https://user-images.githubusercontent.com/50932660/60626126-6290b880-9daf-11e9-855b-3714e6d12b11.png)

And now i've used that with a displacement modifier on a subdivided plane in Blender 2.8 to check that the heightmap is accurate. There, it will accurately recreate the original object.

![unknown (1)](https://user-images.githubusercontent.com/50932660/60626274-db901000-9daf-11e9-9968-722ce28ee838.png)

**Comment by Zylann** [comment]
Can you better explain which difference there is? The screenshots obviously show some differences if taken litterally, but I don't understand what is the nature of that difference.

**Comment by Moonwards** [comment]
This is the node setup used in blender to create that heightmap. I keep looking at it, and it really seems like it should create 309 shades of grey between black and white with that setup. I'm doing what you suggest now but wanted to show the nodes.

![image](https://user-images.githubusercontent.com/50932660/65821074-6db08480-e1f6-11e9-9678-7b7231618020.png)

**Comment by Zylann** [comment]
Yeah, it's likely that color space transformation was applied if you output your values to a `surface` port, which is supposed to be a color.

**Comment by Moonwards** [comment]
![image](https://user-images.githubusercontent.com/50932660/65821533-67bda200-e1fc-11e9-9d01-4e1a8bcf3b51.png)

Pretty much what you predicted. The histogram on the right is of the render from the file on the left in GIMP.

I'm talking to someone and i'm about to ask about it on Blender.SE. Maybe they have an idea of how to correct for it. I don't know how i'm going to export it without using a material output node, so the algorithm seems like it will need to be compensated for or undone.

**Comment by Zylann** [comment]
Maybe fiddling with options here could help?
![image](https://user-images.githubusercontent.com/1311555/65821593-82f5d500-e22f-11e9-924a-5115c9bbb21d.png)

**Comment by Moonwards** [comment]
Here is a file from TheDuriel that has that adjusted for this sort of situation (Standard in the View Transform field instead of Filmic, ) and a node setup for the heightmap generation that adjusts the grayscale. It is an improvement but there are still problems. 

[Duriel_Heightmap_Nodes.zip](https://github.com/Zylann/godot_heightmap_plugin/files/3666095/Duriel_Heightmap_Nodes.zip)

**Comment by Zylann** [comment]
What do I need to check? This is just a Blender file.

**Comment by Moonwards** [comment]
This needs a solution from the Blender side and this is our best shot at it. Anyone else who has created a landscape in Blender and wants to use that with this addon is going to run into this, so it seemed worth posting. Even if it doesn't really solve it. This needs more input from people who know Blender.

**Comment by Zylann** [comment]
The Cycles material editor was built from the ground up to generate colors and displacement to be rendered on screen, not surprising that this problem rises when generating something else.
Either Blender needs to be configured to avoid any kind of color management, or you need to add a node to undo the color space.

My plugin expects correct heightmaps, but maybe a workaround could be done to perform this "undo" in the import function, but I don't know which kind of operation to run here. Godot has `srgb_to_linear` but it doesn't work with HDR images.

**Comment by Zylann** [comment]
Is this still an issue? Is there a feature request to consider? The plugin still just reads the image as it comes without any particular conversion.

**Comment by Moonwards** [comment]
Sorry to get back to you late. I thought your answer before was sufficient, and didn't think it called for a reply.  I got guidance about setting up the nodes correctly in Blender, in a question on blender.se - here https://blender.stackexchange.com/a/154900/7548 if that might be useful to you.

It does seem like it would tend to come up for people who are trying to import something from Blender. Having an option to switch the mode by which the heightmap is interpreted seems like it would be useful. What the conversion should be, I don't know, but it's probably a simple formula.
